src/hotspot/share/opto/loopnode.hpp

```diff
@@ -1630,6 +1630,42 @@ class PhaseIdealLoop : public PhaseTransform {
   // same block.  Split thru the Region.
   void do_split_if(Node *iff, RegionNode** new_false_region = nullptr, RegionNode** new_true_region = nullptr);
 
+private:
+  // Class to keep track of wins in split_through_phi.
+  class SplitWins {
+  private:
+    int _total_wins;
+    int _loop_entry_wins;
+    int _loop_back_wins;
+
+  public:
+    SplitWins() :
+      _total_wins(0),
+      _loop_entry_wins(0),
+      _loop_back_wins(0) {};
```

Comment:
Can you describe somewhere what the definition of these is?
I'm struggling a little with understanding the conditions in `profitable`.